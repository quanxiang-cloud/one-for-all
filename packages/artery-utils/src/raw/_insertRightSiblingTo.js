import byArbitrary from './byArbitrary';
import parent from './parent';
import _insertChildAt from './_insertChildAt';

function _insertRightSiblingTo(root, idOrKeyPath, node) {
  const referenceKeyPath = byArbitrary(root, idOrKeyPath);
  if (!referenceKeyPath) {
    return;
  }

  const referenceNodeIndex = referenceKeyPath.last();
  if (typeof referenceNodeIndex !== 'number') {
    return;
  }

  const parentKeyPath = parent(root, referenceKeyPath);
  return _insertChildAt(root, parentKeyPath, referenceNodeIndex + 1, node);
}

export default _insertRightSiblingTo;
