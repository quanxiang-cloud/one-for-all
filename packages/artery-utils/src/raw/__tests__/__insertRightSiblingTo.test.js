import { fromJS } from 'immutable';
import _insertRightSiblingTo from '../_insertRightSiblingTo';
import NODE from './fixtures/first-level-concrete-children';

// test('_insertRightSiblingTo_to_end', () => {
//   const rootNode = fromJS(NODE);

//   const newRoot = _insertRightSiblingTo(rootNode, 'todo-filter', { foo: 'bar '});

//   expect(newRoot.toJS()).toMatchSnapshot();
// })

test('_insertRightSiblingTo_to_middle', () => {
  const rootNode = fromJS(NODE);

  const newRoot = _insertRightSiblingTo(rootNode, 'todo-list-loop-composedNode', { foo: 'bar '});

  expect(newRoot.toJS()).toMatchSnapshot();
})
