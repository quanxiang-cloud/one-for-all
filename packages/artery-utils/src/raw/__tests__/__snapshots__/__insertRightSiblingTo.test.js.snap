// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_insertRightSiblingTo_to_middle 1`] = `
Object {
  "children": Array [
    Object {
      "id": "ref-schema",
      "schemaID": "whatever",
      "type": "ref-node",
    },
    Object {
      "id": "todo-input-container",
      "name": "div",
      "type": "html-element",
    },
    Object {
      "id": "todo-list-loop-composedNode",
      "node": Object {
        "id": "compose-node-container",
        "nodes": Array [
          Object {
            "id": "todo-toggle",
            "name": "input",
            "type": "html-element",
          },
          Object {
            "id": "todo-title",
            "name": "span",
            "type": "html-element",
          },
          Object {
            "id": "todo-delete",
            "name": "button",
            "type": "html-element",
          },
        ],
        "type": "composed-node",
      },
      "type": "loop-container",
    },
    Object {
      "foo": "bar ",
    },
    Object {
      "children": Array [
        Object {
          "exportName": "TodoFilter",
          "id": "todo-filter",
          "packageName": "todo-app",
          "packageVersion": "whatever",
          "type": "react-component",
        },
      ],
      "id": "footer",
      "name": "div",
      "type": "html-element",
    },
  ],
  "id": "container",
  "name": "div",
  "type": "html-element",
}
`;
